> This LGTM @IEvangelist
> 
> You can :shipit: when ready.

Thanks @BillWagner, but I cannot merge it as the _snippets-build_ is failing. Any ideas on how to address that? These errors seem unrelated:

```
Error:  D:\a\docs\docs>dotnet build "D:\a\docs\docs\samples\snippets\csharp\VS_Snippets_Misc\tpl_parallel\cs\simpleforeach.csproj"  Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET Copyright (C) Microsoft Corporation. All rights reserved.    Determining projects to restore...   Restored D:\a\docs\docs\samples\snippets\csharp\VS_Snippets_Misc\tpl_parallel\cs\simpleforeach.csproj (in 106 ms). D:\a\docs\docs\samples\snippets\csharp\VS_Snippets_Misc\tpl_parallel\cs\parallel_file.cs(51,20): error CA2208: Call the ArgumentException constructor that contains a message and/or paramName parameter [D:\a\docs\docs\samples\snippets\csharp\VS_Snippets_Misc\tpl_parallel\cs\simpleforeach.csproj] D:\a\docs\docs\samples\snippets\csharp\VS_Snippets_Misc\tpl_parallel\cs\simpleparallelfor.cs(124,48): warning CA1416: This call site is reachable on all platforms. 'Console.WindowWidth.set' is only supported on: 'windows'. [D:\a\docs\docs\samples\snippets\csharp\VS_Snippets_Misc\tpl_parallel\cs\simpleforeach.csproj]  Build FAILED.  D:\a\docs\docs\samples\snippets\csharp\VS_Snippets_Misc\tpl_parallel\cs\simpleparallelfor.cs(124,48): warning CA1416: This call site is reachable on all platforms. 'Console.WindowWidth.set' is only supported on: 'windows'. [D:\a\docs\docs\samples\snippets\csharp\VS_Snippets_Misc\tpl_parallel\cs\simpleforeach.csproj] D:\a\docs\docs\samples\snippets\csharp\VS_Snippets_Misc\tpl_parallel\cs\parallel_file.cs(51,20): error CA2208: Call the ArgumentException constructor that contains a message and/or paramName parameter [D:\a\docs\docs\samples\snippets\csharp\VS_Snippets_Misc\tpl_parallel\cs\simpleforeach.csproj]     1 Warning(s)     1 Error(s)  Time Elapsed 00:00:11.12
Error:  D:\a\docs\docs>dotnet build "D:\a\docs\docs\samples\snippets\visualbasic\VS_Snippets_Misc\tpl_parallel\vb\simpleforeach.vbproj"  Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET Copyright (C) Microsoft Corporation. All rights reserved.    Determining projects to restore...   Restored D:\a\docs\docs\samples\snippets\visualbasic\VS_Snippets_Misc\tpl_parallel\vb\simpleforeach.vbproj (in 86 ms). D:\a\docs\docs\samples\snippets\visualbasic\VS_Snippets_Misc\tpl_parallel\vb\fileiteration08.vb(54,50): error CA2208: Call the ArgumentException constructor that contains a message and/or paramName parameter [D:\a\docs\docs\samples\snippets\visualbasic\VS_Snippets_Misc\tpl_parallel\vb\simpleforeach.vbproj] D:\a\docs\docs\samples\snippets\visualbasic\VS_Snippets_Misc\tpl_parallel\vb\simpleparallelfor.vb(104,52): warning CA1416: This call site is reachable on all platforms. 'Public Shared Overloads Property Set WindowWidth(value As Integer)' is only supported on: 'windows'. [D:\a\docs\docs\samples\snippets\visualbasic\VS_Snippets_Misc\tpl_parallel\vb\simpleforeach.vbproj]  Build FAILED.  D:\a\docs\docs\samples\snippets\visualbasic\VS_Snippets_Misc\tpl_parallel\vb\simpleparallelfor.vb(104,52): warning CA1416: This call site is reachable on all platforms. 'Public Shared Overloads Property Set WindowWidth(value As Integer)' is only supported on: 'windows'. [D:\a\docs\docs\samples\snippets\visualbasic\VS_Snippets_Misc\tpl_parallel\vb\simpleforeach.vbproj] D:\a\docs\docs\samples\snippets\visualbasic\VS_Snippets_Misc\tpl_parallel\vb\fileiteration08.vb(54,50): error CA2208: Call the ArgumentException constructor that contains a message and/or paramName parameter [D:\a\docs\docs\samples\snippets\visualbasic\VS_Snippets_Misc\tpl_parallel\vb\simpleforeach.vbproj]     1 Warning(s)     1 Error(s)  Time Elapsed 00:00:01.96
```

--- a/samples/snippets/csharp/VS_Snippets_Misc/tpl_parallel/cs/parallel_file.cs
+++ b/samples/snippets/csharp/VS_Snippets_Misc/tpl_parallel/cs/parallel_file.cs
@@ -48,7 +48,8 @@ class Program
       Stack<string> dirs = new Stack<string>();
 
       if (!Directory.Exists(root)) {
-             throw new ArgumentException();
+            throw new ArgumentException(
+                "The given root directory doesn't exist.", nameof(root));
       }
       dirs.Push(root);
 

--- a/samples/snippets/csharp/VS_Snippets_Misc/tpl_parallel/cs/simpleforeach.csproj
+++ b/samples/snippets/csharp/VS_Snippets_Misc/tpl_parallel/cs/simpleforeach.csproj
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
   </PropertyGroup>
   
 </Project>

--- a/samples/snippets/csharp/VS_Snippets_Misc/tpl_parallel/cs/simpleparallelfor.cs
+++ b/samples/snippets/csharp/VS_Snippets_Misc/tpl_parallel/cs/simpleparallelfor.cs
@@ -2,7 +2,8 @@
 // <snippet01>
  using System;
  using System.Diagnostics;
- using System.Threading.Tasks;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
 
  class MultiplyMatrices
  {
@@ -121,7 +122,12 @@
          Console.Error.WriteLine(c);
          if (Char.ToUpperInvariant(c) == 'Y')
          {
-             if (! Console.IsOutputRedirected) Console.WindowWidth = 180;
+            if (!Console.IsOutputRedirected &&
+                RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                Console.WindowWidth = 180;
+            }
+
              Console.WriteLine();
              for (int x = 0; x < rowCount; x++)
              {

--- a/samples/snippets/visualbasic/VS_Snippets_Misc/tpl_parallel/vb/fileiteration08.vb
+++ b/samples/snippets/visualbasic/VS_Snippets_Misc/tpl_parallel/vb/fileiteration08.vb
@@ -13,25 +13,25 @@ Module Example
         Try
             TraverseTreeParallelForEach("C:\Program Files",
                                         Sub(f)
-                                         ' Exceptions are No-ops.         
-                                         Try
-                                             ' Do nothing with the data except read it.
-                                             Dim data() As Byte = File.ReadAllBytes(f)
-                                             ' In the event the file has been deleted.
-                                         Catch e As FileNotFoundException
+                                            ' Exceptions are No-ops.         
+                                            Try
+                                                ' Do nothing with the data except read it.
+                                                Dim data() As Byte = File.ReadAllBytes(f)
+                                                ' In the event the file has been deleted.
+                                            Catch e As FileNotFoundException
 
-                                             ' General I/O exception, especially if the file is in use.
-                                         Catch e As IOException
+                                                ' General I/O exception, especially if the file is in use.
+                                            Catch e As IOException
 
-                                             ' Lack of adequate permissions.
-                                         Catch e As UnauthorizedAccessException
+                                                ' Lack of adequate permissions.
+                                            Catch e As UnauthorizedAccessException
 
-                                             ' Lack of adequate permissions.
-                                         Catch e As SecurityException
+                                                ' Lack of adequate permissions.
+                                            Catch e As SecurityException
 
                                             End Try
-                                         ' Display the filename.
-                                         Console.WriteLine(f)
+                                            ' Display the filename.
+                                            Console.WriteLine(f)
                                         End Sub)
         Catch e As ArgumentException
             Console.WriteLine("The directory 'C:\Program Files' does not exist.")
@@ -51,7 +51,8 @@ Module Example
         ' Data structure to hold names of subfolders to be examined for files.
         Dim dirs As New Stack(Of String)
 
-        If Not Directory.Exists(root) Then Throw New ArgumentException()
+        If Not Directory.Exists(root) Then Throw New ArgumentException(
+            "The given root directory doesn't exist.", NameOf(root))
 
         dirs.Push(root)
 

--- a/samples/snippets/visualbasic/VS_Snippets_Misc/tpl_parallel/vb/simpleforeach.vbproj
+++ b/samples/snippets/visualbasic/VS_Snippets_Misc/tpl_parallel/vb/simpleforeach.vbproj
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
   </PropertyGroup>
   
 </Project>

--- a/samples/snippets/visualbasic/VS_Snippets_Misc/tpl_parallel/vb/simpleparallelfor.vb
+++ b/samples/snippets/visualbasic/VS_Snippets_Misc/tpl_parallel/vb/simpleparallelfor.vb
@@ -1,5 +1,6 @@
 ﻿' <snippet01>
 Imports System.Diagnostics
+Imports System.Runtime.InteropServices
 Imports System.Threading.Tasks
 
 Module MultiplyMatrices
@@ -101,7 +102,9 @@ Module MultiplyMatrices
         Dim c As Char = Console.ReadKey(True).KeyChar
         Console.Error.WriteLine(c)
         If Char.ToUpperInvariant(c) = "Y"c Then
-            If Not Console.IsOutputRedirected Then Console.WindowWidth = 168
+            If Not Console.IsOutputRedirected AndAlso
+                RuntimeInformation.IsOSPlatform(OSPlatform.Windows) Then Console.WindowWidth = 168
+
             Console.WriteLine()
             For x As Integer = 0 To rowCount - 1
                 Console.WriteLine("ROW {0}: ", x)

